cfme/tests/configure/test_log_depot_operation.py

```diff
@@ -183,12 +196,88 @@ def check_ftp(ftp, server_name, server_zone_id):
             )
 
 
+@pytest.fixture(scope="module")
+def wait_for_ansible(appliance):
+    appliance.server.settings.enable_server_roles("embedded_ansible")
+    appliance.wait_for_embedded_ansible()
+    yield
+    appliance.server.settings.disable_server_roles("embedded_ansible")
+
+
+@pytest.fixture(scope="module")
+def ansible_repository(appliance, wait_for_ansible):
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://github.com/quarckster/ansible_playbooks",
+        description=fauxfactory.gen_alpha())
+    view = navigate_to(repository, "Details")
+    if appliance.version < "5.9":
+        refresh = view.browser.refresh
+    else:
+        refresh = view.toolbar.refresh.click
+    wait_for(
+        lambda: view.entities.summary("Properties").get_text_of("Status") == "successful",
+        timeout=60,
+        fail_func=refresh
+    )
+    yield repository
+
+    if repository.exists:
+        repository.delete()
+
+
+@pytest.fixture(scope="module")
+def ansible_catalog_item(appliance, ansible_repository):
+    cat_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.ANSIBLE_PLAYBOOK,
+        fauxfactory.gen_alphanumeric(),
+        fauxfactory.gen_alphanumeric(),
+        display_in_catalog=True,
+        provisioning={
+            "repository": ansible_repository.name,
+            "playbook": "dump_all_variables.yml",
+            "machine_credential": "CFME Default Credential",
+            "create_new": True,
+            "provisioning_dialog_name": fauxfactory.gen_alphanumeric(),
+            "extra_vars": [("some_var", "some_value")]
+        }
+    )
+    yield cat_item
+
+    if cat_item.exists:
+        cat_item.delete()
+
+
+@pytest.fixture(scope="module")
+def catalog(appliance, ansible_catalog_item):
+    catalog_ = appliance.collections.catalogs.create(fauxfactory.gen_alphanumeric(),
+                                                     description='my catalog',
+                                                     items=[ansible_catalog_item.name])
+    ansible_catalog_item.catalog = catalog_
+    yield catalog_
+
+    if catalog_.exists:
+        catalog_.delete()
+        ansible_catalog_item.catalog = None
+
+
+@pytest.fixture
+def service_request(appliance, ansible_catalog_item):
+    request_descr = "Provisioning Service [{0}] from [{0}]".format(ansible_catalog_item.name)
```

Comment:
Optional:
In such cases, I will prefer to use `key` than `index` like
`request_descr = "Provisioning Service [{name}] from [{name}]".format(name=ansible_catalog_item.name)`

Reply:
yep, looks better, I changed